File: FlutterNodeBackend/models/Blog.js
import mongoose from "mongoose";
const { Schema } = mongoose;

// Define the comment schema
const commentSchema = new Schema({
    text: { type: String, required: true },
    user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    createdAt: { type: Date, default: Date.now }
});

// Define the blog schema
const blogSchema = new Schema({
    title: { type: String, required: true },
    description: { type: String, required: true },
    comments: [{ type: Schema.Types.ObjectId, ref: 'Comment' }],  // Reference to Comment model
    user: { type: Schema.Types.ObjectId, ref: 'User', required: true }  // Reference to the User who created the blog
}, { timestamps: true });

// Create models for Comment and Blog
const Comment = mongoose.model('Comment', commentSchema);
const Blog = mongoose.model('Blog', blogSchema);

export { Comment, Blog };

File: lib/entities/Blog.dart
import 'Comment.dart';

class Blog {
  final String id;
  final String title;
  final String description;
  final String content;
  final String userId;
  final List<Comment> comments;

  Blog({
    required this.id,
    required this.title,
    required this.description,
    required this.content,
    required this.userId,
    required this.comments,
  });

  factory Blog.fromJson(Map<String, dynamic> json) {
    return Blog(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      content: json['content'] ?? '',
      userId: json['user']?['_id'] ?? '',
      comments: (json['comments'] as List<dynamic>?)
          ?.map((comment) => Comment.fromJson(comment))
          .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'title': title,
      'description': description,
      'content': content,
      'user': userId,
      'comments': comments.map((comment) => comment.toJson()).toList(),
    };
  }
}
